Shared/UI/Face Detection/FaceImageCaptureViewController.swift

```diff
@@ -123,11 +155,101 @@ extension FaceImageCaptureViewController: AVCaptureVideoDataOutputSampleBufferDe
             guard let maskPixelBuffer
                     = self.segmentationRequest.results?.first?.pixelBuffer else { return }
             // Process the images.
-            self.blend(original: imageBuffer, mask: maskPixelBuffer)
+            let blendedImage = self.blend(original: imageBuffer, mask: maskPixelBuffer)
 
+            // Set the new, blended image as current.
+            self.currentCIImage = blendedImage
         } catch {
             logError(error)
         }
+
+        switch self.videoCaptureState {
+        case .idle:
+            // Do nothing
+            break
+        case .starting:
+            // Initialize the AVAsset writer to prepare for capture
+            self.initializeAssetWriter(with: sampleBuffer)
+            self.videoCaptureState = .capturing
+        case .capturing:
+            self.writeSampleToFile(sampleBuffer)
+        case .ending:
+            self.finishWritingVideo()
+            self.videoCaptureState = .idle
+        }
+    }
+
+    private func initializeAssetWriter(with sampleBuffer: CMSampleBuffer) {
+        do {
+            // Get a url to temporarily store the video
+            #warning("Use a full UUID")
+            let uuid = UUID().uuidString.prefix(8)
+            let url = URL(fileURLWithPath: NSTemporaryDirectory(),
+                          isDirectory: true).appendingPathComponent(uuid+".mov")
+
+            // Create an asset writer that will write the video the url
+            self.videoWriter = try AVAssetWriter(outputURL: url, fileType: .mov)
+            let settings: [String : Any] = [AVVideoCodecKey : AVVideoCodecType.hevc,
+                                            AVVideoWidthKey : 720,
+                                           AVVideoHeightKey : 720,
+                            AVVideoCompressionPropertiesKey : [AVVideoQualityKey : 0.5]]
+
+            self.videoWriterInput = AVAssetWriterInput(mediaType: AVMediaType.video,
+                                                       outputSettings: settings)
+            self.videoWriterInput.mediaTimeScale = CMTimeScale(bitPattern: 600)
+            self.videoWriterInput.expectsMediaDataInRealTime = true
+
+            self.pixelBufferAdaptor
+            = AVAssetWriterInputPixelBufferAdaptor(assetWriterInput: self.videoWriterInput,
+                                                   sourcePixelBufferAttributes: nil)
+
+            if self.videoWriter.canAdd(self.videoWriterInput) {
+                self.videoWriter.add(self.videoWriterInput)
+            }
+
+            self.videoWriter.startWriting()
+
+            let startTime = CMSampleBufferGetPresentationTimeStamp(sampleBuffer)
+            self.videoWriter.startSession(atSourceTime: startTime)
+            self.videoStartTime = CMSampleBufferGetPresentationTimeStamp(sampleBuffer).seconds
```

Comment:
Image data flows like this:
ImageData -> AVAssetWriterInputPixelBufferAdaptor -> AVAssetWriterInput -> AVAssetWriter -> Disk

This code sets up that pipeline so we can start video capture.